File: BTVN_18112025/src/main/java/com/example/homework1/StudentAdapter.java
package com.example.homework1;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class StudentAdapter extends RecyclerView.Adapter<StudentAdapter.ViewHolder> {

    Context context;
    ArrayList<Student> list;
    ItemClickListener listener;

    public interface ItemClickListener {
        void onItemClick(int position);
        void onDeleteClick(int position);
    }

    public StudentAdapter(Context context, ArrayList<Student> list, ItemClickListener listener) {
        this.context = context;
        this.list = list;
        this.listener = listener;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_student, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Student s = list.get(position);
        holder.txtName.setText(s.name);
        holder.txtMssv.setText(s.mssv);

        holder.itemView.setOnClickListener(v ->
                listener.onItemClick(position)
        );

        holder.btnDelete.setOnClickListener(v ->
                listener.onDeleteClick(position)
        );
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView txtName, txtMssv;
        ImageView btnDelete;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            txtName = itemView.findViewById(R.id.txtName);
            txtMssv = itemView.findViewById(R.id.txtMssv);
            btnDelete = itemView.findViewById(R.id.btnDelete);
        }
    }
}

File: BTVN_16122025/app/src/main/java/com/example/studentmanager/StudentAdapter.kt
package com.example.studentmanager

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class StudentAdapter(
    private var students: MutableList<Student>,
    private val onItemClick: (Student) -> Unit
) : RecyclerView.Adapter<StudentAdapter.StudentViewHolder>() {

    class StudentViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val tvName: TextView = view.findViewById(R.id.tvName)
        val tvId: TextView = view.findViewById(R.id.tvId)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StudentViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_student, parent, false)
        return StudentViewHolder(view)
    }

    override fun onBindViewHolder(holder: StudentViewHolder, position: Int) {
        val student = students[position]
        holder.tvName.text = student.name
        holder.tvId.text = student.id

        holder.itemView.setOnClickListener {
            onItemClick(student)
        }
    }

    override fun getItemCount() = students.size

    fun updateData(newStudents: MutableList<Student>) {
        students = newStudents
        notifyDataSetChanged()
    }
}